src/vs/workbench/contrib/interactive/browser/dummyFileSystemProvider.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter, Event } from 'vs/base/common/event';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { URI } from 'vs/base/common/uri';
+import { FileSystemProviderCapabilities, FileType, IFileChange, IFileDeleteOptions, IFileOpenOptions, IFileOverwriteOptions, IFileSystemProvider, IFileWriteOptions, IStat, IWatchOptions } from 'vs/platform/files/common/files';
+
+export class DummyFileSystem implements IFileSystemProvider {
+	private fdCounter = 1;
+
+	capabilities: FileSystemProviderCapabilities = FileSystemProviderCapabilities.FileOpenReadWriteClose;
+
+	public readFile(_resource: URI): Promise<Uint8Array> {
+		return Promise.resolve(new Uint8Array);
+	}
+
+	public open(_resource: URI, _opts: IFileOpenOptions): Promise<number> {
+		return Promise.resolve(this.fdCounter++);
+	}
+
+	public stat(_resource: URI): Promise<IStat> {
+		return Promise.resolve({
+			type: FileType.File,
+			ctime: 0,
+			mtime: 0,
+			size: 0,
+		});
+	}
+	public mkdir(_resource: URI): Promise<void> {
+		return Promise.resolve();
+	}
+
+	public readdir(_resource: URI): Promise<[string, FileType][]> {
+		return Promise.resolve([['', FileType.Unknown]]);
+	}
+
+	public delete(_resource: URI, _opts: IFileDeleteOptions): Promise<void> {
+		return Promise.resolve();
+	}
+
+	public rename(_from: URI, _to: URI, _opts: IFileOverwriteOptions): Promise<void> {
+		return Promise.resolve();
+	}
+
+	copy?(_from: URI, _to: URI, _opts: IFileOverwriteOptions): Promise<void> {
```

Comment:
Can you omit some of these optional methods?

Reply:
I was hitting 'method does not exist' errors, so I just added them all. But I was able to drop a few that were not necessary.